lib/node.ts

```diff
@@ -34,5 +37,6 @@ export function attachToPage(page: any, url: string, rules: AutoCMP[], retries =
   }
   page.on('framenavigated', onFrame);
   page.frames().forEach(onFrame);
-  return new TabConsent(tab, detectDialog(tab, retries, rules))
+  await prehideElements(tab, rules);
```

Comment:
I would make this optional - i.e. the caller specifies if prehiding should be done or not (as it's an invasive change).